File: apis/src/main/java/com/beat/apis/performance/application/dto/create/PerformanceImageResponse.java
package com.beat.apis.performance.application.dto.create;

import com.beat.global.support.jackson.CdnImageUrl;

public record PerformanceImageResponse(
	Long imageId,
	@CdnImageUrl String imageUrl
) {
	public static PerformanceImageResponse of(
		Long imageId,
		String imageUrl
	) {
		return new PerformanceImageResponse(
			imageId,
			imageUrl
		);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/PerformanceImageResponse.kt
package com.beat.apis.performance.api.response

import com.beat.application.frontoffice.performance.PerformanceImageResult
import com.beat.global.support.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class PerformanceImageResponse private constructor(
    val imageId: Long?, @field:CdnImageUrl val imageUrl: String?,
) {
    companion object {
        fun from(result: PerformanceImageResult): PerformanceImageResponse = PerformanceImageResponse(result.id, result.image)
    }
}
